gdal_subsetter/transform.py

```diff
@@ -251,32 +279,39 @@ def process_netcdf(self, source, basename, input_filename, output_dir, layername
 
         return layernames, result, process_msg
 
-    def process_zip(self, source, basename, input_filename, output_dir, layernames):
-
-        result = None
-
-        variables = source.process('variables')
-
-        variableslist = [item.name for item in variables]
-
-        [is_tif, tiffile, msg_tif, is_nc, ncfile, msg_nc] = self.unpack_zipfile(input_filename, output_dir, variableslist)
+    def process_zip(self, source: HarmonySource, basename: str,
+                    zipfile_name: str, output_dir: str, layernames: List[str]):
+        """ Unpack the input zip file and process the individual files it
+            contains. If there are GeoTIFF files, only these will be
+            processed.
 
-        if is_tif:
-            if tiffile:
-                layernames, result = self.process_geotiff(
-                    source, basename, tiffile, output_dir, layernames)
+            `HarmonyAdapter.unpack_zipfile` will attempt to stack multiple
+            GeoTIFFs contained within the zip file, and process those as a
+            single file. The possibility of multiple NetCDF-4 files does not
+            seem to be accounted for here.
 
-            return layernames, result, msg_tif
-
-        elif is_nc:
-            if ncfile:
-                layernames, result = self.process_netcdf(source, basename,
-                                                         ncfile, output_dir,
-                                                         layernames)
-
-            return layernames, result, msg_nc
+        """
+        variable_names = [item.name for item in source.process('variables')]
+        tiffile, msg_tif, ncfile, msg_nc = self.unpack_zipfile(zipfile_name,
+                                                               output_dir,
+                                                               variable_names)
+
+        if len(tiffile) > 0:
+            layernames, result = self.process_geotiff(source, basename,
+                                                      tiffile, output_dir,
+                                                      layernames)
+            message = msg_tif
+        elif len(ncfile) > 0:
+            layernames, result = self.process_netcdf(source, basename,
+                                                     ncfile, output_dir,
+                                                     layernames)
+            message = msg_nc
+        else:
+            result = None
+            message = ('The granule does not have GeoTIFF or NetCDF-4 files, '
+                       'will not process.')
```

Comment:
I would much prefer to just raise an exception here, and similarly in `HarmonyAdapter.unpack_zipfile`, instead of propagating error messages around to later raise them. I thought I'd hold off and see if you (Matt) agreed with that.

If that sounds good, I think there are a number of places that would benefit from that clean-up.

Reply:
Okay - I'll make that change in its own PR (the next one in this series), and I'll update the regression tests at the same time.